src/snowflake/connector/auth/keypair.py

```diff
@@ -63,22 +68,27 @@ def __init__(self, private_key, lifetime_in_seconds: int = LIFETIME):
                 "JWT_CNXN_RETRY_ATTEMPTS", AuthByKeyPair.DEFAULT_JWT_RETRY_ATTEMPTS
             )
         )
-        self._jwt_cnxn_wait_time = timedelta(
+        self._timeout = timedelta(
             seconds=int(
                 os.getenv(
                     "JWT_CNXN_WAIT_TIME", AuthByKeyPair.DEFAULT_JWT_CNXN_WAIT_TIME
                 )
             )
-        )
+        ).total_seconds()
         self._current_retry_count = 0
 
-    def authenticate(
+    def reset_secrets(self) -> None:
+        self._private_key = None
+
+    @property
+    def type_(self) -> AuthType:
+        return AuthType.KEY_PAIR
+
+    def prepare(
         self,
-        authenticator: str,
-        service_name: str | None,
         account: str,
         user: str,
-        password: str | None,
+        **kwargs: Any,
     ) -> str:
```

Comment:
should the signature also be keyword arguments to be consistent with the protocol?
```python
def prepare(
    self,
    *,
    authenticator: str,
    service_name: str | None,
    account: str,
    user: str,
    password: str | None,
    **kwargs
)